qiskit_ibm_runtime/quantum_program/quantum_program.py

```diff
@@ -138,14 +139,13 @@ def __init__(
         samplex: Samplex,
         *,
         samplex_arguments: dict[str, np.ndarray | PauliLindbladMap] | None = None,
-        shape: tuple[int, ...] | None = None,
+        samplex_shape: tuple[int, ...] | None = None,
         chunk_size: int | None = None,
     ):
-        if not isinstance(circuit, QuantumCircuit):
-            raise ValueError(f"Expected {repr(circuit)} to be a QuantumCircuit.")
+        super().__init__(circuit=circuit, chunk_size=chunk_size)
```

Comment:
We usually don't verify argument types. But this is done anyway inside `super().__init__`.